tests/test_recordio.py

```diff
@@ -0,0 +1,74 @@
+import json
+
+from dcos import recordio
+from dcos.errors import DCOSException
+
+
+def test_encode():
+    try:
+        encoder = recordio.Encoder(json.dumps)
+    except Exception as exception:
+        raise DCOSException("Error instantiating 'RecordIO' encoder: {error}"
+                            .format(error=exception))
+
+    try:
+        message = {
+            "type": "ATTACH_CONTAINER_OUTPUT",
+            "containerId": "123456789"
+        }
+
+        encoded = encoder.encode(message)
+
+    except Exception as exception:
+        raise DCOSException("Error encoding 'RecordIO' message: {error}"
+                            .format(error=exception))
+
+    string = json.dumps(message)
+    assert encoded == str(len(string)) + "\n" + string
+
+
+def test_encode_decode():
+    total_messages = 10
+
+    try:
+        encoder = recordio.Encoder(json.dumps)
+    except Exception as exception:
+        raise DCOSException("Error instantiating 'RecordIO' encoder: {error}"
+                            .format(error=exception))
+
+    try:
+        decoder = recordio.Decoder(json.loads)
+    except Exception as exception:
+        raise DCOSException("Error instantiating 'RecordIO' decoder: {error}"
+                            .format(error=exception))
+
+    try:
+        message = {
+            "type": "ATTACH_CONTAINER_OUTPUT",
+            "containerId": "123456789"
+        }
+
+        encoded = ""
+        for i in range(total_messages):
+            encoded += encoder.encode(message)
+
+    except Exception as exception:
+        raise DCOSException("Error encoding 'RecordIO' message: {error}"
+                            .format(error=exception))
+
+    try:
+        all_records = []
+        offset = 0
+        chunk_size = 5
+        while offset < len(encoded):
+            records = decoder.decode(encoded[offset:offset + chunk_size])
```

Comment:
Then you just return `encoded` from offset:<end> (that's what python does by default)